cmd/tsdbrelay/main.go

```diff
@@ -104,4 +115,65 @@ func (rp *relayProxy) ServeHTTP(responseWriter http.ResponseWriter, r *http.Requ
 		resp.Body.Close()
 		verbose("bosun relay success")
 	}()
+	if parse && denormalizationRules != nil {
+		go rp.denormalize(&reader.buf)
```

Comment:
Use `bytes.NewReader(reader.buf.Bytes())` here, as above, because otherwise the reader position of the buffer could be affected by or affect other go routines.